src/org/intellij/erlang/completion/ErlangAppCompletionContributor.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2012-2015 Sergey Ignatov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.intellij.erlang.completion;
+
+import com.intellij.codeInsight.completion.*;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.patterns.PatternCondition;
+import com.intellij.patterns.PsiElementPattern;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.ProcessingContext;
+import com.intellij.util.containers.ContainerUtil;
+import org.intellij.erlang.ErlangFileType;
+import org.intellij.erlang.psi.*;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+import static com.intellij.patterns.PlatformPatterns.psiElement;
+import static com.intellij.patterns.PlatformPatterns.virtualFile;
+import static com.intellij.patterns.StandardPatterns.instanceOf;
+
+public class ErlangAppCompletionContributor extends CompletionContributor {
+  // source: "FILE SYNTAX" section of http://www.erlang.org/doc/man/app.html
+  static final List<String> KEYWORDS =
+    ContainerUtil.immutableList("description", "id", "vsn", "modules", "maxP", "maxT", "registered", "applications",
+                                "included_applications", "env", "mod", "start_phases", "runtime_dependencies");
+  private static final String APPLICATION_KEYWORD = "application";
+
+  @Override
+  public void beforeCompletion(@NotNull CompletionInitializationContext context) {
+    PsiFile file = context.getFile();
+    if (file.getFileType() != ErlangFileType.APP) return;
+    int startOffset = context.getStartOffset();
+    PsiElement elementAt = file.findElementAt(startOffset);
+    PsiElement parent = elementAt != null ? elementAt.getParent() : null;
+    if (parent instanceof ErlangTupleExpression) {
+      context.setDummyIdentifier(APPLICATION_KEYWORD);
+    }
+  }
+
+  public ErlangAppCompletionContributor() {
+    PsiElementPattern.Capture<PsiElement> psiElementInAppFile =
+      psiElement().inFile(instanceOf(ErlangFile.class)).inVirtualFile(virtualFile().ofType(ErlangFileType.APP));
```

Comment:
I cannot write here something like `psiElement(ErlangAtom.class)` because addCompletions is waiting for leafPsiElement to edit its text. We need to stay at the lowest position in the tree.